### ISSUE TITLE
Factory method to get a new `app` object

### ISSUE THREAD

**Issue by poswald** [django-webtest/django-webtest#5]
Thanks for the project! I have a quick feature request. Perhaps there is a way to do it already... It would be nice to have a factory method to get a new `app` object that is set up the same way as the `self.app` object.

I often have to check that two users can interact in a sane way (user A creates an object, user B can view it) and my functional test would need two logged in users. In these cases I use `self.app` for one of them, and would need to set up a second one because the session is tied to `self.app.session` but I want them to be set up identically (and with all of the nice things) django-webtest provides.

What would be the best way to go about doing that? Let me know what you think. Thanks!

**Comment by codeinthehole** [comment]
I think this is only tangentially relevant, but here's how I use webtest as a pytest fixture:

```python
import pytest
from django_webtest import DjangoTestApp, WebTestMixin

@pytest.fixture(scope='function')
def webtest(request):
    """
    Provide the "app" object from WebTest as a fixture

    Taken and adapted from https://gist.github.com/magopian/6673250
    """
    # Patch settings on startup
    wtm = WebTestMixin()
    wtm._patch_settings()

    # Unpatch settings on teardown
    request.addfinalizer(wtm._unpatch_settings)

    return DjangoTestApp()
```

**Comment by merwok** [comment]
See also #60 with suggested code that’s a bit different: the WebTestMixin is created only once.

In my setup I’m using a session-scoped fixture instead of a global object; I’ll update the ticket once I work around #64.
